docs/future-api.md

```diff
@@ -0,0 +1,129 @@
+## mediachain.next api notes
+
+Everything here is in flux and subject to change; this is just a sketch. 
+The idea is to give an overview of the high-level api surface of an "online" node, that is, 
+one that actively contributes to the network and is expected to remain connected to the network.
+
+All the code examples below are in javascript, with flow-style annotations for return types.
+
+### Peer ids & local state
+A peer has a unique id, which is the hash of their public key.  They may also have a collection of 
+certificates, signed by moderators of the namespaces they're authorized to participate in.
+
+The peer-local state includes the peer's id, certificate store, private key, etc.  It also includes the 
+index of connected peers, which allows us to route operations to peers with the correct roles.
+
+For the examples below, the peer-local state is assumed to exist in an accesible place 
+(like an object instance variable, etc), so it's omitted from the function arguments.
+
+### Domain-specific ids (Well-Known-Identifiers)
+The RFC-4 design uses the term Well-Known-Identifier (WKI) to refer to an existing domain specific id, 
+for example `nypl:510d47e2-ef28-a3d9-e040-e00a18064a99` which refers to an object in the New York Public Library collection.
+
+**question:** is there a shared directory of WKI prefixes? How does a user know to use `getty` instead 
+of e.g. `gettyimages`, or that `getty` doesn't refer to http://www.getty.edu/?
+
+### Connect / peer discovery
+Ideally a single `connect()` method that optionally allows you to specify the address of a bootstrap peer / 
+directory server. If not provided, picks a peer from a hard-coded set built into the app. 
+
+```javascript
+function connect(options): Promise<Unit> {
+  let {bootstrapAddress} = options
+  // ...
+}
+```
+
+*mock design*: This is a noop, as we're mocking a fully connected node with a direct route to all (virtual) peers.
+
+*p2p design*: Contacts a directory server to update the peer index and register any services we're capable of providing. 
+May push updates that we've spooled locally to any peers that were previously subscribed to us.  
+May register for periodic updates from the director server.
+
+### Publication
+A node can publish to any leaf namespace that it has write permission for.  If a namespace is not specified, statements 
+are published to the user's personal namespace, which corresponds to the hash of their peer id, e.g. `universe.peers.QmF00123` 
+
+**questions:** should we accept a list of namespaces, to publish to multiple namespaces at once?
```

Comment:
at API level, sure, but let's keep it a list of 1 for now to minimize concerns like atomicity etc